django/forms/boundfield.py

```diff
@@ -130,6 +131,21 @@ def value(self):
             data = self.field.bound_data(self.data, data)
         return self.field.prepare_value(data)
 
+    def _did_change(self):
+        field = self.field
+        if field.show_hidden_initial:
+            hidden_widget = field.hidden_widget()
+            try:
+                initial_value = field.to_python(
+                    self.form._widget_data_value(hidden_widget, self.html_initial_name)
+                )
```

Comment:
I think that `ValidationError` is only raised out of `Field.to_python()`, not `Widget.value_from_datadict()`, so this can be simplified:

```suggestion
            widget = field.hidden_widget()
            value = self.form._widget_data_value(widget, self.html_initial_name)
            try:
                initial_value = field.to_python(value)
```